src/server/routes/register.ts

```diff
@@ -77,19 +87,26 @@ router.post(
 
 router.post(
   Routes.REGISTRATION,
+  recaptcha.middleware.verify,
   handleAsyncErrors(async (req: Request, res: ResponseWithRequestState) => {
     let state = res.locals;
-
     const { email = '' } = req.body;
-    const { password = '' } = req.body;
-
-    const { returnUrl } = state.queryParams;
 
+    const { returnUrl, ref, refViewId } = state.queryParams;
     try {
-      await create(email, password, req.ip);
-      // TODO: Can we remove this second call to get cookies for the user once we move over to Okta?
-      const cookies = await authenticate(email, password, req.ip);
-      setIDAPICookies(res, cookies);
+      if (req.recaptcha?.error) {
+        logger.error(
+          'Problem verifying recaptcha, error response: ',
+          req.recaptcha.error,
+        );
+        throw {
+          message: CaptchaErrors.GENERIC,
```

Comment:
Throw an error if server-side recaptcha verification fails. This is shown on the client side so the user is aware;